serverside/src/event/event.service.ts

```diff
@@ -1,4 +1,171 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { Class, Event, PrismaClient, Teacher, Test } from '@prisma/client';
+import { createEventDto } from 'dto/createEventDto';
 
 @Injectable()
-export class EventService {}
+export class EventService {
+  async deleteEventByID(id: number, prisma: PrismaClient): Promise<any> {
+    return prisma.event.delete({
+      where: {
+        eventID: id,
+      },
+    });
+  }
+  async manipulateEventByID(
+    id: number,
+    newEvent: createEventDto,
+    prisma: PrismaClient,
+  ): Promise<any> {
+    //check for valid teacher, class, test
+    await validInput(newEvent, prisma);
+
+    return prisma.event.update({
+      data: newEvent,
+      where: {
+        eventID: id,
+      },
+    });
+  }
+
+  async getEventsByTeacherID(
+    teacherID: number,
+    timespan: number,
+    prisma: PrismaClient,
+  ): Promise<Event[]> {
+    const now = new Date();
+    const toDate = new Date(
+      now.getFullYear() + (timespan || 1),
```

Comment:
Timespan lieber in Monaten übergeben, falls es sehr viele Events in einem Jahr gibt